### PR TITLE
dev/core#2814 Fix activity:sendEmail to use renderTemplate

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#2814 Fix activity:sendEmail to use renderTemplate

Before
----------------------------------------
`ReplaceContactTokens` called

After
----------------------------------------
`renderTemplate` called

Technical Details
----------------------------------------
There is good test cover on this. I added an extra test for the scenario where smarty is enabled in this context. All other changes in the relevant tests are tidy ups & don't materially change them.

Note that this does create a situation where more queries could occur but
1) this is only called in core from the email task - which is limited to 50 contacts so it
is low-volume

2) this is really the start of a cleanup - there are other places where
queries can be removed when we look at the total flow but I feel we need to get
some simplification happening first & once we are doing things in a more consistent way I will be better placed to rationalise the queries

3) this seems to be called in extensions to some extent - and it should still work
- there might be more queries but I think the 'support contract' for calling an
unsupported function does not include it working in ways not required by
core code

Comments
----------------------------------------

### CODE REVIEW

File: tests/phpunit/CRM/Activity/BAO/ActivityTest.php [hunk 1]
```diff
@@ -1655,36 +1655,57 @@ public function testSendEmailWillReplaceTokensUniquelyForEachContact() {
...
+      [
+        $contactID => [
+          'preferred_mail_format' => 'Both',
+          'contact_id' => $contactID,
+          'email' => 'a@example.com',
+        ],
+      ],
+      '{contact.first_name} {$contact.first_name}',
+      '{contact.first_name} {$contact.first_name}',
+     '{contact.first_name} {$contact.first_name}',
```
Comment (demeritcowboy) [inline tests/phpunit/CRM/Activity/BAO/ActivityTest.php:1672]: phpcs doesn't catch these indents I guess. Also down on line 1732.

Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Activity/BAO/ActivityTest.php]: That's not like phpcs - to be forgiving - fixed

**Comment by colemanw** [comment]
I agree test cover is good, and this PR adds more tests. Refactoring makes sense.

**Comment by demeritcowboy** [comment]
Emails now have the literal `{contact.XXX}` in them.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy yeah - looks like the tests focussed on the created activity & I didn't spot it wasn't passing through. Fix coming

**Pull request merged**